### PR TITLE
Add support for Mullvad Browser

### PR DESCRIPTION
**Pull request by fyqwbdzsfjh**
As discussed in https://github.com/Kicksecure/tb-updater/issues/25

Notes:
- Download, verification and installation works (tested only cli).
- https://cdn.mullvad.net/browser/update_responses/update_1/alpha/downloads.json reports `13.5a1` as the alpha version, but only `13.5a2` is available at https://dist.torproject.org/mullvadbrowser/
- Wiki page needs to be created
- `tb_kill_already_running_tb_maybe` needs to kill `mullvadbrowser.real`
- `tb_preparation` has to give an error if $tb_onion = true, as an onion for `TBB_VERSIONS_FILE_LINK` does not seem to exist
- Does not start the browser, as tb-starter does not support it yet
- `mb` has the possibility to conflict with something else a possible user has. Using the full name would be better in my opinion, but this is consistent with the other scripts.

Feel free to ask any changes (I am quite comfortable with bash).

### CODE REVIEW

File: usr/bin/update-torbrowser [hunk 1]
```diff
@@ -897,6 +897,10 @@ tb_preparation() {
          [ -n "$CURL_FORCE_SSL" ] || CURL_FORCE_SSL="--tlsv1.3 --proto =https --cert-status"
       fi
 
+      if [ -z "$TBB_VERSIONS_FILE_LINK" ] && [ "$tb_browser_name" = "mullvad-browser" ]; then
+         TBB_VERSIONS_FILE_LINK="https://cdn.mullvad.net/browser/update_responses/update_1/${TBB_RELEASE_CHANNEL}/downloads.json"
```
Comment (adrelanos) [inline usr/bin/update-torbrowser:901]: Could you make this a variables too please?

The variable parts are:

* `https://cdn.mullvad.net/browser/update_responses/update_1`
* `https://aus1.torproject.org/torbrowser/update_3`

These could be split into two:

* `https://cdn.mullvad.net`
* `https://aus1.torproject.org`

and:

* `browser/update_responses/update_1`
* `torbrowser/update_3`

Then also clearnet vs onion needs to be considered.

**Comment by adrelanos** [comment]
Looks nice and clean. Left 1 inline comment.

Wiki page created:
https://www.kicksecure.com/wiki/Mullvad_Browser

> mb has the possibility to conflict with something else a possible user has. Using the full name would be better in my opinion, but this is consistent with the other scripts.

Right. I am not adamant about consistency here keeping the super shot folder names. Full name seem better indeed.

(Because upon reflection even `tb` / `.tb` seems wrong from my perspective nowadays but that's a different and unrelated issue now that you mention it.)

**Comment by fyqwbdzsfjh** [comment]
Changed the folder name from "mb" to "mullvadbrowser". The folder ends up being `$HOME/.mullvadbrowser/mullvad-browser/` now.

Should a new variable `$tb_process_name` be defined and set to `firefox.real` by default (and `mullvadbrowser.real` in update-mullvadbrowser)?

Also noticed that Mullvad Browser package does not contain `tb_local_version_file`-like file, which is slightly problematic. At a quick glance there does not appear to be a file that contains the installed version.

**Comment by adrelanos** [comment]
> Changed the folder name from "mb" to "mullvadbrowser". The folder ends up being `$HOME/.mullvadbrowser/mullvad-browser/` now.

Great!

> Should a new variable `$tb_process_name` be defined and set to `firefox.real` by default (and `mullvadbrowser.real` in update-mullvadbrowser)?

Yes, please. That seems clean.

> Also noticed that Mullvad Browser package does not contain `tb_local_version_file`-like file, which is slightly problematic. At a quick glance there does not appear to be a file that contains the installed version.

Could you please report this bug upstream?

> Also, the wiki base url is hardcoded to be https://www.whonix.org/wiki/. Should that be made to a variable as well?

That would be good.

**Comment by fyqwbdzsfjh** [comment]
> > Should a new variable `$tb_process_name` be defined and set to `firefox.real` by default (and `mullvadbrowser.real` in update-mullvadbrowser)?
> 
> Yes, please. That seems clean.
> 

Done

> > Also noticed that Mullvad Browser package does not contain `tb_local_version_file`-like file, which is slightly problematic. At a quick glance there does not appear to be a file that contains the installed version.
> 
> Could you please report this bug upstream?
> 

I've requested an account to Tor Project's GitLab and will do so when it's accepted.

I'm pretty sure these lines are responsible for that file
https://gitlab.torproject.org/tpo/applications/tor-browser-build/-/blob/main/projects/browser/build#L305-308

> > Also, the wiki base url is hardcoded to be https://www.whonix.org/wiki/. Should that be made to a variable as well?
> 
> That would be good.

On a further glance, there are quite many links to Whonix's wiki and replacing them all will not work. Should only the links that refer to `$tb_wiki`-page be replaced?

**Comment by adrelanos** [comment]
Yes. That would suffice. The links aren't a blocker for this to be merged.

**Comment by adrelanos** [comment]
For gitlab account in case there are issues, try `gitlab-admin@torproject.org`.

This looks nice already! Does it work? Ready for merge?

**Comment by fyqwbdzsfjh** [comment]
Version file merged.

https://gitlab.torproject.org/tpo/applications/tor-browser-build/-/issues/41044

**Comment by adrelanos** [comment]
Other stuff if you're interested in contributing... To ensure this doesn't break in the future due to refactoring, changes and go unnoticed would be adding github actions / CI testing.

**Pull request merged**